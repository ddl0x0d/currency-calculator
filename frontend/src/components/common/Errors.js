import Alert from 'react-bootstrap/Alert';
import AppContext from '../../AppContext';
import Col from 'react-bootstrap/Col';
import PropTypes from 'prop-types';
import React, { useContext } from 'react';
import Row from 'react-bootstrap/Row';

const propTypes = {
  messages: PropTypes.array.isRequired
};

const Errors = props => {
  const context = useContext(AppContext);

  const hide = () => {
    context.showErrors([]);
  };

  return (
    <Row className="justify-content-center">
      <Col md={8} xl={4}>
        <Alert
          className="small"
          dismissible={true}
          show={props.messages.length > 0}
          onClose={hide}
          variant="danger"
        >
          {props.messages.map(error => (
            <div key={props.messages.indexOf(error)}>
              {error.field || ''} {error.message}
            </div>
          ))}
        </Alert>
      </Col>
    </Row>
  );
};

Errors.propTypes = propTypes;

export default Errors;
