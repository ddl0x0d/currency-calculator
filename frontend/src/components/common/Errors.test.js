import Errors from './Errors';
import React from 'react';
import { mount } from 'enzyme';

describe('<Errors />', () => {
  const errors = messages => <Errors messages={messages} />;

  it('renders correctly without errors', () => {
    const wrapper = mount(errors([]));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with errors', () => {
    const messages = [
      { field: 'Field', message: 'has error' },
      { message: 'There is an error' }
    ];
    const wrapper = mount(errors(messages));
    expect(wrapper).toMatchSnapshot();
  });
});
