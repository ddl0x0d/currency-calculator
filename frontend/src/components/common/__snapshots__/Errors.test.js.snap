// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Errors /> renders correctly with errors 1`] = `
<Errors
  messages={
    Array [
      Object {
        "field": "Field",
        "message": "has error",
      },
      Object {
        "message": "There is an error",
      },
    ]
  }
>
  <Bootstrap(Row)
    className="justify-content-center"
  >
    <Row
      as="div"
      bsPrefix="row"
      className="justify-content-center"
      noGutters={false}
    >
      <div
        className="justify-content-center row"
      >
        <Col
          as="div"
          md={8}
          xl={4}
        >
          <div
            className="col-xl-4 col-md-8"
          >
            <Alert
              className="small"
              closeLabel="Close alert"
              dismissible={true}
              onClose={[Function]}
              show={true}
              transition={[Function]}
              variant="danger"
            >
              <Fade
                appear={false}
                in={true}
                mountOnEnter={false}
                timeout={300}
                unmountOnExit={true}
              >
                <Transition
                  addEndListener={[Function]}
                  appear={false}
                  enter={true}
                  exit={true}
                  in={true}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={300}
                  unmountOnExit={true}
                >
                  <div
                    className="fade small alert alert-danger alert-dismissible show"
                    role="alert"
                  >
                    <CloseButton
                      label="Close alert"
                      onClick={[Function]}
                    >
                      <button
                        className="close"
                        onClick={[Function]}
                        type="button"
                      >
                        <span
                          aria-hidden="true"
                        >
                          Ã—
                        </span>
                        <span
                          className="sr-only"
                        >
                          Close alert
                        </span>
                      </button>
                    </CloseButton>
                    <div
                      key="0"
                    >
                      Field
                       
                      has error
                    </div>
                    <div
                      key="1"
                    >
                       
                      There is an error
                    </div>
                  </div>
                </Transition>
              </Fade>
            </Alert>
          </div>
        </Col>
      </div>
    </Row>
  </Bootstrap(Row)>
</Errors>
`;

exports[`<Errors /> renders correctly without errors 1`] = `
<Errors
  messages={Array []}
>
  <Bootstrap(Row)
    className="justify-content-center"
  >
    <Row
      as="div"
      bsPrefix="row"
      className="justify-content-center"
      noGutters={false}
    >
      <div
        className="justify-content-center row"
      >
        <Col
          as="div"
          md={8}
          xl={4}
        >
          <div
            className="col-xl-4 col-md-8"
          >
            <Alert
              className="small"
              closeLabel="Close alert"
              dismissible={true}
              onClose={[Function]}
              show={false}
              transition={[Function]}
              variant="danger"
            >
              <Fade
                appear={false}
                in={false}
                mountOnEnter={false}
                timeout={300}
                unmountOnExit={true}
              >
                <Transition
                  addEndListener={[Function]}
                  appear={false}
                  enter={true}
                  exit={true}
                  in={false}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={300}
                  unmountOnExit={true}
                />
              </Fade>
            </Alert>
          </div>
        </Col>
      </div>
    </Row>
  </Bootstrap(Row)>
</Errors>
`;
